client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -145,10 +208,19 @@ function mediaButton( editor ) {
 			return;
 		}
 
+		const dirtyImage = MediaStore.getDirtyImage();
+		let numOfImagesToUpdate = 0;
+
 		isVisualEditMode = ! editor.isHidden();
 
 		if ( isVisualEditMode ) {
-			images = editor.dom.select( 'img' );
+			if ( dirtyImage ) {
+				images = editor.dom.select( `img.wp-image-${ dirtyImage.ID }` );
+
+				numOfImagesToUpdate = images.length;
```

Comment:
We need to remember how many dirty images there are because after we update all of them, we need to set the image media `.isDirty` flag to `false` so the images won't be updated every time another image is going to be updated.